neuralprophet/forecaster.py

```diff
@@ -384,32 +385,39 @@ def add_lagged_regressor(self, names, regularization=None, normalize="auto", onl
         ----------
             names : string or list
                 name of the regressor/list of regressors.
+            n_covars : int
+                previous regressors time steps to use as input in the predictor (covar order)
+                if ``auto``, time steps will be equivalent to the AR order (default)
+                if ``scalar``, all the regressors will only use last known value as input
             regularization : float
                 optional  scale for regularization strength
             normalize : bool
                 optional, specify whether this regressor will benormalized prior to fitting.
                 if ``auto``, binary regressors will not be normalized.
-            only_last_value : bool
-                specifies last value handling
-
-                Options
-                    * (default) ``False`` use same number of lags as auto-regression
-                    * ``True`` only use last known value as input
         """
+        if n_covars == "auto":
+            if self.n_lags is not None and self.n_lags > 0:
+                n_covars = self.n_lags
+                log.info("n_covars is equal to n_lags")
+            else:
+                n_covars = 1
+                log.info("n_covars is equal to 1")
+        if n_covars == "scalar":
+            n_covars = 1
+            log.info("n_covars is equal to 1")
+        only_last_value = False if n_covars > 1 else True
         if self.fitted:
             raise Exception("Covariates must be added prior to model fitting.")
-        if self.n_lags == 0:
-            raise Exception("Covariates must be set jointly with Auto-Regression.")
+        if n_covars == 0 or n_covars is None:
+            raise Exception("Please, set number of lags for covariates (n_covars > 0)")
```

Comment:
We could instead log a warning, and just do nothing (return self) -> this would not be a fatal error

Reply:
Ok. However, it breaks the code in the assertion from time_dataset:
(line 264)                assert covar_config[covar].n_lags > 0